designs/CentrallyManagingNuGetPackageVersions-VSManagePackages.md

```diff
@@ -0,0 +1,95 @@
+
+## Who are the customers
+
+All customers managing packages on projects that are using central package version management with projects in Visual Studio IDE.
+
+## Goals
+
+1. When installing a package in a centrally managed project, if the package already exists in the `Directory.Packages.props` file, only the project file is modified. The `PackageReference` element added must not have a version applied.
+2. When installing a package in a centrally managed project, if the package does not already exist in the `Directory.Packages.props` file, then the package version is added to `Directory.Packages.props`, in addition to goal 1 above.
+3. When updating a centrally managed package version, the `Directory.Packages.props` file is updated, and all projects that use the package are restored with the new package version.
+4. When uninstalling/removing a package from a centrally managed project, the `PackageReference` is removed from the project file, but the `PackageVersion` is kept in the `Directory.Packages.props` file.
+
+## Non - Goals
+
+a. This spec is scoped to Visual Studio API interaction between components. Package Manager UI changes are out of scope. Command Line Interface is out of scope.
+b. Removing a `PackageVersion` from the `Directory.Packages.props` file is out of scope for the first version, but may be in scope in the future.
+
+## Scenarios
+
+The following scenarios will be enabled when the proposed design changes will be completed.
+
+1. Installing a package in a project with Package Manager UI or Package Manager Console.
+2. Installing a package from a quick action when editing a source file.
+3. Removing a package from a project with Package Manager UI or Package Manager Console.
+4. Updating the version of a package for all projects using that `Directory.Packages.props` file.
+
+## Requirements
+
+The requirements are defined in the [feature spec](https://github.com/NuGet/Home/wiki/Centrally-managing-NuGet-package-versions).
+
+Note that:
+
+* It's valid for some, not all, projects in the solution to use centrally managed package versions.
+* It's valid for different projects using central package versions to use different `Directory.Packages.props` files.
+* As is valid now, it's valid for some projects to use `packages.config`, while others use `PackageReference`. The `packages.config` projects cannot use centrally managed package versions, and individual `PackageReference` projects may opt in or out of central package version management.
+
+## Solution
+
+### Installing a package in a project
+
+Currently NuGet calls project system APIs to add a PackageReference to the project for the given package id and version. However, when the project uses centrally managed versions, the project file must not add the version on the `PackageReference` element.
```

Comment:
Would be possible to add more information about the Project System APIs used to add/update PackageReferences?

Reply:
In case you haven't yet read to the end of the document, I provided a little detail in annex 1, but I'll add more details.

Reply:
I've added more details to annex 1 now. Let me know if it doesn't contain the information that you're interested in.